packages/app-headless-cms/src/admin/menus/ContentGroupsMenuItems.tsx

```diff
@@ -67,60 +67,75 @@ export const ContentGroupsMenuItems = () => {
         return null;
     }
 
-    return (
-        <>
-            {groups.map(group => {
-                return (
-                    <HasContentEntryPermissions key={group.id} group={group}>
-                        <>
-                            <Menu
-                                name={group.id}
-                                parent={"headlessCMS"}
-                                element={
-                                    <Menu.Group
-                                        text={group.name}
-                                        icon={
-                                            <Menu.Group.Icon
-                                                label={group.name}
-                                                element={<Icon group={group} />}
-                                            />
-                                        }
-                                    />
-                                }
-                            />
+    /**
+     * Renders a menu item for a given content group.
+     * Displays the group's name and icon in the menu.
+     */
+    const GroupMenu = ({ group }: { group: CmsGroup }) => (
+        <Menu
+            name={group.id}
+            parent="headlessCMSContent"
+            element={
+                <Menu.Item
+                    text={group.name}
+                    icon={<Menu.Link.Icon label="Content" element={<Icon group={group} />} />}
+                />
+            }
+        />
+    );
+
+    /**
+     * Renders menu items for all content models within a group.
+     * If the group has no content models, displays a "Nothing to show" message.
+     * Wraps each content model menu item with permission checks.
+     */
+    const GroupContentModels = ({ group }: { group: CmsGroup }) => {
```

Comment:
Same here I'd say. We can even pull these into separate sibling files (idk, maybe `CmsGroupMenuItem` / `CmsModelMenuItems`), just so we don't have everything in a single file.